spec/system/donation_system_spec.rb

```diff
@@ -51,6 +51,19 @@
         expect(page).to have_css("table tbody tr", count: 1)
       end
 
+      it "Filters by diaper drives" do
+        a = create(:diaper_drive, name: 'A')
+        b = create(:diaper_drive, name: "B")
+        x = create(:diaper_drive_participant, business_name: "X")
+        create(:diaper_drive_donation, diaper_drive: a, diaper_drive_participant: x)
+        create(:diaper_drive_donation, diaper_drive: b, diaper_drive_participant: x)
+        visit subject
+        expect(page).to have_css("table tbody tr", count: 2)
+        select a.name, from: "filters_by_diaper_drive"
+        click_button "Filter"
+        expect(page).to have_css("table tbody tr", count: 1)
```

Comment:
Nitpick: one thing that would be good to also test is that the page displays the donation from the desired diaper drive, not just that it only displays one :) That said, this is great and I'll be merging it as-is!